syntax = "proto3";

package master_pb;

//////////////////////////////////////////////

service MasterNode {
    rpc Upload (UploadRequest) returns (UploadResponse) {

    }

    rpc AddVolume (AddVolumeRequest) returns (AddVolumeResponse) {

    }
}

//////////////////////////////////////////////
message UploadRequest {
    string file_path = 1;
    bytes content = 2;
    uint32 file_size = 3;
}

message UploadResponse {
    enum UploadResponseCode {
        unknown = 0;
        success = 1;
        failed = 2;
    }
    uint32 volume_id = 1;
}

message AddVolumeRequest {
    string volume_grpc = 1;
}

message AddVolumeResponse {
    enum AddVolumeResponseCode {
        unknown = 0;
        success = 1;
    }
    uint32 volume_id = 1;
    AddVolumeResponseCode response_code = 2;
}
